### ISSUE TITLE
Issues with AMP

### ISSUE THREAD

**Issue by RauchenwaldC** [o10n-x/wordpress-css-optimization#40]
I'm not sure which of the plugins is the issue, but overall they are not supporting AMP pages.
As you can see at https://validator.ampproject.org/#url=https%3A%2F%2Fwww.infantblog.com%2Fbaby-blog%2Fhow-white-noise-may-help-your-infant%2Famp%2F the custom script is not AMP conform.
For now it should be considered if the page is AMP and in that case disable all functionallity(?)?

**Comment by optimalisatie** [comment]
You are loading the critical CSS via a stylesheet link:

```html
<link data-o10n rel="stylesheet"
href="/wp-content/cache/o10n/css/0/1/1:critical.css"/>
```

The AMP issues were addressed before by @AlphaDE.

In regards to support for custom scripts. It may be best to simply
disable the plugins for AMP pages.

More info is available in the following issues:

https://github.com/o10n-x/wordpress-css-optimization/issues/31
https://github.com/o10n-x/wordpress-javascript-optimization/issues/9

It is important to remember that the optimization plugins don't do much
(WordPress) automation. In regards to AMP compatibility, it will be up
to the user to for example disable or tune it for the intended result.

More control in the hands of the optimization specialist = potential for
better results for any website/environment, today and in the future. The
plugins are 100% JSON controlled, unlocking potential for AI based
automation. AI wouldn't need to know about automation or the (ever
changing) WordPress environment, it would be able to resolve all issues
with JSON based logic. It could resolve AMP related issues within a
simple structured configuration mechanism that would work for any
website and even any CMS.

In regards to simplicity for users, a potential extension that could be
created on the basis of the plugins is a "Optimization Wizard" that
would semi-automatically tune a website to achieve perfect Google
Lighthouse scores and the ultimate performance result. It could use
'sensors' such as automated cross-browser javascript error testing (a
new innovation, does not yet exist), test suites such as Page Speed and
Google Analytics for in-depth analysis of a website's performance issues
and much more. This would enable a AI to 'see'. The JSON would be it's
fingers. A potential cloud base AI service could use the 'intelligence'
created on thousands of websites to automatically achieve the ultimate
performance result for any HTML page for any user (device, network
condition etc, essentially considering millions of conditions in real
time to provide the ultimate performance result for any user in any
condition).

So in short: JSON offers both exceptional value (time saving etc) for
optimization specialists and for potential AI based automation.

We are currently looking for a party to take over the project.

**Comment by RauchenwaldC** [comment]
Thanks for your reply. #31 solved it :)
